staging/src/k8s.io/apiextensions-apiserver/pkg/controller/openapi/builder.go

```diff
@@ -390,18 +388,15 @@ func newBuilder(crd *apiextensions.CustomResourceDefinition, version string, sch
 	if crd.Spec.Scope == apiextensions.NamespaceScoped {
 		b.namespaced = true
 	}
-	// Pre-build schema with Kubernetes native properties
-	if schema != nil {
-		b.schema = b.buildKubeNative(schema)
-	} else {
-		b.schema.AddExtension(endpoints.ROUTE_META_GVK, []map[string]string{
-			{
-				"group":   b.group,
-				"version": b.version,
-				"kind":    b.kind,
-			},
-		})
+
+	if schema == nil {
+		schema = &spec.Schema{
+			SchemaProps: spec.SchemaProps{Type: []string{"object"}},
+		}
 	}
+
+	// Pre-build schema with Kubernetes native properties
+	b.schema = b.buildKubeNative(schema)
```

Comment:
adding fields to empty schema closes the door for unknown fields during client-side validation

we cannot do this before we honor `additionalProperties` in kube-openapi and kubectl

Reply:
if we add `additionalProperties: true` explicitly, that would work, right?

But very good observation. Thanks for noticing that.

Reply:
Which reminds me again that we have to stop using heuristics to make specs stricter. We can leave out interpretations, but never ever be stricter than OpenAPI is officially.

Reply:
my understanding is that we don't honor `additionalProperties: true` in this [code path](https://github.com/kubernetes/kube-openapi/blob/ea82251f3668f8c1bde607fa6e20e5bf36e576a4/pkg/util/proto/document.go#L237-L238)

> if we add `additionalProperties: true` explicitly, that would work, right?

I will double check

Reply:
We don't. I checked already.